packages/flutter/lib/src/material/expansion_panel.dart

```diff
@@ -329,11 +329,17 @@ class _ExpansionPanelListState extends State<ExpansionPanelList> {
           widget.expansionCallback!(childIndex, false);
         }
       }
+        //Because when _handlePressed the state is not yet expanded
+      widget.expansionCallback?.call(index, !isExpanded);
```

Comment:
Does this line actually set isExpanded to false for the selected child expansion? If yes, why is this line necessary? What would be the behavior if this line is removed?

Reply:
It does not set isExpanded to false. is Expanded is false for the widget at this point because it is about to be open when setState is called so we pass the opposite of isExpanded because it will be expanded as the callback fires. If this line is commented out there will never be a call notifiying that a panel is open. Only the callback of the previously open panel closing will be called.